contracts/main.py

```diff
@@ -1,105 +1,121 @@
+            message = self.error_message.token_undefined()
+        )
+        from_user = self.ledger_key.make(current_from, params.token_id)
+        sp.verify(
+            (self.data.ledger[from_user].balance >= 1),
+            message = self.error_message.insufficient_balance())
+        to_user = self.ledger_key.make(params.to_, params.token_id)
+        self.data.ledger[from_user].balance = sp.as_nat(
+            self.data.ledger[from_user].balance - 1)
+        sp.if self.data.ledger.contains(to_user):
+            self.data.ledger[to_user].balance += 1
+        sp.else:
+                self.data.ledger[to_user] = FA2.Ledger_value.make(1)
+                
 
-    @sp.entry_point
-    def sell(self, params):
-        sp.verify(sp.mutez(0) <= params.price)
-        self.checkAvailable(self.data.hounds[params.houndId], params)
-        self.data.hounds[params.houndId].price = params.price
+class Hound(sp.Contract):
+    def __init__(self, contract_address):
+        self.init(hounds = {},counter = 0, contract_address = contract_address)
 
     @sp.entry_point
-    def lend(self, params):
-        sp.verify(sp.mutez(0) <= params.price)
-        self.checkAvailable(self.data.hounds[params.houndId], params)
-        self.data.hounds[params.houndId].borrowPrice = params.price
-
+    def createHound(self, params):
+        sp.set_type(params.hound.generation, sp.TInt)
+        sp.set_type(params.hound.genome, sp.TString)
+        token_contract = sp.contract(sp.TRecord(creator = sp.TAddress, metadata = sp.TMap(sp.TString, sp.TBytes),token_id = sp.TNat ), self.data.contract_address, entry_point = "mint").open_some()
+        sp.transfer(sp.record(creator = sp.sender, token_id = self.data.counter, metadata = FA2.FA2_token_metadata.make_metadata(
+        decimals = 0,
+        name = 'Hound NFT',
+        symbol = 'Hound'
+        )), sp.mutez(0), token_contract)
+        self.data.hounds[self.data.counter] = sp.record(token_id = self.data.counter, owner = sp.sender, creator = sp.sender, genome = params.hound.genome, isNew = True, timestamp = params.hound.timestamp, generation = params.hound.generation)
+        self.data.counter+=1
+    
     @sp.entry_point
-    def buy(self, params):
-        hound = self.data.hounds[params.houndId]
-        sp.verify(sp.mutez(0) < hound.price)
-        sp.verify(hound.price <= params.price)
-        sp.verify(sp.amount == params.price)
-        sp.send(hound.owner, params.price)
-        hound.owner = sp.sender
-        sp.if hound.isNew:
-            hound.isNew = False
-            hound.auction = sp.now.add_seconds(self.newAuctionDuration)
-        sp.verify(sp.now <= hound.auction)
-        sp.if sp.now <= hound.auction:
-            hound.price = params.price + sp.mutez(1)
+    def breed(self, params):
+        sp.verify(params.p1!=params.p2)
+        sp.set_type(params.p1, sp.TNat)
+        sp.set_type(params.p2, sp.TNat)
+        g1 = self.data.hounds[params.p1].genome

```

Comment:
can we use more descriptive variable names here?
instead of `p1`, `p2`, maybe use `parent1, parent2`.
`g1` -> `genome1`, `g2` -> `genome2` etc.